ming_sim/db.py

```
@@ -535,6 +535,72 @@ def _has_stop_condition(stop_condition: object) -> bool:
+
+
+def directive_payload_has_ordinary_triad(payload: Mapping[str, object]) -> bool:
+    """普通旨意结构化 triad：动作类型 + 目标类型 + 目标ID。"""
+    return all(
+        str(payload.get(key) or "").strip()
+        for key in ("dossier_action_type", "target_kind", "target_id")
+    )
+
+
+def directive_payload_admits_structured_write(payload: Mapping[str, object]) -> bool:
+    """旨意可落库：普通 triad，或 #658 御笔强推 typed target。"""
```

Ruff (0.16.2)

[warning] Docstring contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF002)